src/schema-association-service.ts

```diff
@@ -237,12 +261,48 @@ async function autoDetectSchema(
     if (session === undefined) {
         logger.log(`No organization found for ${workspaceFolder.name}`, 'SchemaDetection');
         vscode.window.showErrorMessage(format(Messages.unableToAccessOrganization, organizationName));
+        
+        // Disable 1ESPT schema and delete 1ESPT schema file if user is signed out
+        config.update('1ESPipelineTemplatesSchemaFile', undefined, vscode.ConfigurationTarget.Workspace); // disable 1ESPT schema
+        logger.log("1ESPT schema disabled as user is not signed in", 'SchemaDetection')
+        // show message to user that 1ESPT schema is disabled as user is not signed in
+        vscode.window.showInformationMessage(Messages.disabled1ESPTSchemaAsUserNotSignedInMessage)
+        
         return undefined;
     }
 
     // Create the global storage folder to guarantee that it exists.
     await vscode.workspace.fs.createDirectory(context.globalStorageUri);
 
+    // Try to fetch schema in the following order:
+    // 1. Cached 1ESPT schema
+    // 2. 1ESPT schema if user is signed in with microsoft account and has enabled 1ESPT schema
+    // 3. Cached Organization specific schema
+    // 4. Organization specific schema
+
+    // get cached 1ESPT schema if:
+    // 1. User is signed in with microsoft account
+    // 2. 1ESPT schema is enabled
+    // 3. 1ESPT schema is not older than 24 hours
+    const [schemaUri1ESPT , skipOrgSpecificCachedSchema] = await getCached1ESPTSchemaInformation(context, organizationName, session, oneesptSchemaEnabled, lastUpdated1ESPTSchema, seen1ESPTOrganizations);
```

Comment:
If the config setting is false, why do we still try to get the cached information anyway?
The logic in that function is a little complex at the moment so I'm trying to simplify it - it looks like we'll always return `[undefined, false]` if `oneesptSchemaEnabled` is false, which means we'll skip everything from L288 to L304 - perfect for wrapping all of that in an `if (oneesptSchemaEnabled)` block :).

Reply:
done. Also, simplified the logic to get cached schema